ast/TreeEquality.cc

```diff
@@ -345,9 +350,37 @@ bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const Exp
         return false;
     }
 
-    return structurallyEqual(gs, avoid, tree.tag(), tree.get(), other.get(), file, root);
+    return traverse<Comparator>(gs, avoid, tree.tag(), tree.get(), other.get(), file, root);
 }
 
+struct StructurallyEqualComparator {
+    static bool compareNodes(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree,
+                             const ExpressionPtr &other, const core::FileRef file) {
+        return traverse<StructurallyEqualComparator>(gs, avoid, tree, other, file);
+    }
+
+    static bool compareSpans(const core::GlobalState &gs, const void *avoid, absl::Span<const ExpressionPtr> a,
+                             absl::Span<const ExpressionPtr> b, const core::FileRef file) {
+        return traverseSpan<StructurallyEqualComparator>(gs, avoid, a, b, file);
+    }
+};
+
+struct ExactlyEqualComparator {
```

Comment:
Could you add a TODO here to remove this once the prism migration is complete? We should also add that to the migration issue.

Reply:
Might be good to refer back to #9065, so that anyone looking at in the future has a point of reference.